pkg/agent/flexvolume/crd/types.go

```diff
@@ -42,8 +42,16 @@ type Attachment struct {
 	ReadOnly     bool   `json:"readOnly"`
 }
 
-type VolumeattachmentList struct {
+type VolumeAttachmentList struct {
 	metav1.TypeMeta `json:",inline"`
 	metav1.ListMeta `json:"metadata"`
-	Items           []Volumeattachment `json:"items"`
+	Items           []VolumeAttachment `json:"items"`
+}
+
+// VolumeAttachmentControllerInterface handles custom resource VolumeAttachment storage operations
```

Comment:
nice way to handle it with a different controller instead of trying to implement it differently throughout the code.